llama_index/indices/managed/vectara/base.py

```diff
@@ -0,0 +1,234 @@
+"""Managed index.
+
+A managed Index - where the index is accessible via some API that interfaces a managed service.
+
+"""
+
+import os
+import logging
+import requests
+import json
+
+from typing import Any, Optional, List
+from llama_index.indices.managed.base import ManagedIndex
+from llama_index.schema import Document
+from llama_index.indices.base_retriever import BaseRetriever
+from llama_index.indices.managed.vectara.retriever import VectaraRetriever
+from llama_index.data_structs.data_structs import IndexStruct, IndexStructType
+
+_logger = logging.getLogger(__name__)
+
+
+class VectaraIndexStruct(IndexStruct):
+    """Vectara Index Struct."""
+
+    @classmethod
+    def get_type(cls) -> IndexStructType:
+        """Get index struct type."""
+        return IndexStructType.VECTARA
+
+
+class VectaraIndex(ManagedIndex):
+    """Vectara Index.
+
+    The Vectara index implements a managed index that uses Vectara as the backend.
+    Vectara performs a lot of the functions in traditional indexes in the backend:
+    - breaks down a document into chunks (nodes)
+    - Creates the embedding for each chunk (node)
+    - Performs the search for the top k most similar nodes to a query
+    - Optionally can perform summarization of the top k nodes
+
+    Args:
+        show_progress (bool): Whether to show tqdm progress bars. Defaults to False.
+
+    """
+
+    def __init__(
```

Comment:
I think this init function is still missing a way to initialize with nodes.

Both `VectaraIndex(nodes)` or `VectaraIndex.from_documents(documents)` should both be possible I think (and the attached notebook should likely also use `from_documents` 👍🏻)

I think `_build_index_from_nodes` needs to be implemented for those two entry points to work?

Reply:
So here is where I may not fully understand things about Llama-index, so could use your guidance.
1. With Vectara (as a managed service) we break the text in the backend into chunks (which I understand would be nodes in LI). My idea here was that you don't do this during indexing (it gets done at the back-end of the API / managed service) but is exposed as nodes during query time
2. If you do VectaraIndex(nodes) is a node simple the exact same as a document (i.e. not a chunk but the full text)? This would not be maintained as such as Vectara woudl break it down into real chunks in the backend anyway
3. re: from_documents() - I see it's implemented in BaseIndex, but again assumes nodes (thus your comment on _build_index_from_nodes), which is not exactly the case here. 
Can you please advise what's the best path given the LI philosophy and design?

Reply:
1. Right, this makes sense
2. Yea, the Node and Index classes are essentially the same. I wouldn't recommend inserting already chunked nodes in most cases, but there would also he nothing stopping someone from doing so? 🤔 maybe if a user had large node chunks or something, idk 😅
3. `from_documents` could be overridden to insert the documents themselves, just to maintain API similarity?

So, it seems like `VectaraIndex(nodes)` and `VectaraIndex.from_documents(documents)` could be implemented to use the same underlying function to throw things into vectara? 

Let me know if this makes sense!

Reply:
yes. Please see my update and let me know if this is what you had in mind in terms of using nodes and from_documents()